proto/wippersnapper/signal/v1/signal.proto

```diff
@@ -7,176 +7,86 @@ package wippersnapper.signal.v1;
-    wippersnapper.i2c.v1.I2CDeviceDeinitRequest req_i2c_device_deinit       = 5;
-    wippersnapper.i2c.v1.I2CDeviceUpdateRequest req_i2c_device_update       = 6;
-    wippersnapper.i2c.v1.I2CDeviceInitRequests req_i2c_device_init_requests = 7;
+    // digitalio.proto
+    wippersnapper.digitalio.v1.DigitalIOAdd digitalio_add = 1;
+    wippersnapper.digitalio.v1.DigitalIOAdd digitalio_remove = 2;
+    wippersnapper.digitalio.v1.DigitalIOEvent digitalio_event = 3;
+    // analogio.proto
+    wippersnapper.analogio.v1.AnalogIOAdd analogio_add = 4;
+    wippersnapper.analogio.v1.AnalogIOAdd analogio_remove = 5;
+    // description.proto
+    wippersnapper.description.v1.CreateDescription create_description = 6;
+    // servo.proto
+    wippersnapper.servo.v1.ServoAdd servo_add = 7;
+    wippersnapper.servo.v1.ServoRemove servo_remove = 8;
+    wippersnapper.servo.v1.ServoWrite servo_write   = 9;
+    // pwm.proto
+    wippersnapper.pwm.v1.PWMAdd pwm_add                        = 10;
+    wippersnapper.pwm.v1.PWMRemove pwm_remove                        = 11;
+    wippersnapper.pwm.v1.PWMWriteDutyCycle pwm_write_duty            = 12;
+    wippersnapper.pwm.v1.PWMWriteDutyCycleMulti pwm_write_duty_multi = 13;
+    wippersnapper.pwm.v1.PWMWriteFrequency pwm_write_freq            = 14;
+    // pixels.proto
+    wippersnapper.pixels.v1.PixelsAdd pixels_add = 15;
+    wippersnapper.pixels.v1.PixelsRemove pixels_remove = 16;
+    wippersnapper.pixels.v1.PixelsWrite pixels_write   = 17;
+    // ds18x20.proto
+    wippersnapper.ds18x20.v1.Ds18x20Add ds18x20_add     = 18;
+    wippersnapper.ds18x20.v1.Ds18x20Remove ds18x20_remove = 19;
+    // uart.proto
+    wippersnapper.uart.v1.UARTAdd uart_add = 20;
+    wippersnapper.uart.v1.UARTRemove uart_remove = 21;
+    // i2c.proto
+    wippersnapper.i2c.v1.I2CScan i2c_scan = 22;
+    wippersnapper.i2c.v1.I2CInits i2c_add = 23;
+    wippersnapper.i2c.v1.I2CRemove i2c_remove = 24;
+    // error.proto
+    wippersnapper.error.v1.Error error = 25;
   }
 }
 
-/**
-* I2CResponse represents the device's response to an I2C-specific message from IO.
-*/
-message I2CResponse {
+message DeviceToBroker {
   option (nanopb_msgopt).submsg_callback = true;
   oneof payload {
-    wippersnapper.i2c.v1.I2CBusInitResponse resp_i2c_init                  = 1 [deprecated = true, (nanopb).type = FT_IGNORE];
-    wippersnapper.i2c.v1.I2CBusScanResponse resp_i2c_scan                  = 2;
-    wippersnapper.i2c.v1.I2CDeviceInitResponse resp_i2c_device_init        = 3;
-    wippersnapper.i2c.v1.I2CDeviceDeinitResponse resp_i2c_device_deinit    = 4;
-    wippersnapper.i2c.v1.I2CDeviceUpdateResponse resp_i2c_device_update    = 5;
-    wippersnapper.i2c.v1.I2CDeviceEvent resp_i2c_device_event              = 6;
+    //digitalio.proto
+    wippersnapper.digitalio.v1.DigitalIOEvent digitalio_event = 1;
+    // analogio.proto
+    wippersnapper.analogio.v1.AnalogIOEvent analogio_event = 2;
```

Comment:
Are we avoiding an `*Added` event for io components? I forget if we talked about that.

Reply:
I am not sure, what are your thoughts?

Reply:
I don't know the hardware side of course, but if you can definitely tell if you've successfully connected the selected component, then yeah I'd advocate we have `AnalogIOAdded` and `DigitalIOAdded` messages to fit the pattern and improve user experience.

Reply:
@lorennorman  To clarify here - we want a **new** message called AnalogIOAdded. We are not changing `AnalogIOAdd` to AnalogIOAdded`.